crates/red_knot/src/types/infer.rs

```diff
@@ -119,12 +120,30 @@ where
     }
 }
 
+fn get_class_member<Db>(db: &Db, class_id: ClassTypeId, name: &Name) -> QueryResult<Option<Type>>
```

Comment:
This isn't a query at all (i.e. it's not exposed as part of any Db trait), it's just a free function that happens to take a db as an argument. (The same is currently true of `infer_expr_type`, though that one almost certainly will become a real query.)

This is not directly cached because it's just a thin convenience wrapper over `infer_symbol_type`, which is cached.

Whether this is a free function or a method on `ClassTypeId` is just a question of which API we prefer, it's not related to whether or not it's a query (a free function can just be a regular function, not a query), so it's not related to salsa either.

But I do think having it be a method on `ClassTypeId` is nice, in particular because it means we don't need a `FileId` on `ClassType` anymore, nor an extra `file_id` argument. I'll make that change.